### PR TITLE
Add modal filtering capability to Mirgecom

### PR DESCRIPTION
**Pull request by thomasgibson**
This PR revives all the hard work by @MTCam in #92 (and also supersedes #92). Uses some recent functionality in meshmode and grudge.

### CODE REVIEW

File: mirgecom/filter.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
+""":mod:`mirgecom.filter` is for filters and filter-related constructs.
```
Comment (inducer) [inline mirgecom/filter.py:1]: This should have an `.. automodule::` somewhere so that it's included in the docs.

File: mirgecom/filter.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            mode = sum(mode_id)
+            if mode >= cutoff:
+                filter_mat[mode_index, mode_index] = \
+                    mode_response_function(mode,
+                                           cutoff=cutoff,
+                                           nfilt=nfilt)
+
+        return actx.freeze(actx.from_numpy(filter_mat))
+
+    return _spectral_filter_matrix(group)
```
Comment (inducer) [inline mirgecom/filter.py:104]: This does way more flops than needed. (i.e. no need for a matrix if you're just wanting to scale stuff)

**Review by inducer — COMMENTED** [review]
Thanks! Two quick notes here.

File: mirgecom/filter.py (hunk 2 above)
Comment (thomasgibson) [reply mirgecom/filter.py:104]: @inducer Ah, right. So how about we just build a scaling vector? Then that will simplify the `apply_filter_matrix` method (won't actually do a matvec, just element-wise multiply)

Comment (inducer) [reply mirgecom/filter.py:104]: Yep, exactly.

File: mirgecom/filter.py (hunk 1 above)
Comment (thomasgibson) [reply mirgecom/filter.py:1]: Done: https://github.com/illinois-ceesd/mirgecom/pull/324/commits/b6002f806d01c7a0df9b2b1775c2d81044005301

File: mirgecom/filter.py (hunk 2 above)
Comment (thomasgibson) [reply mirgecom/filter.py]: Done: https://github.com/illinois-ceesd/mirgecom/pull/324/commits/ae92e70c602bcd084e4b034287883de953be0f85

File: mirgecom/filter.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
+""":mod:`mirgecom.filter` is for filters and filter-related constructs.
+
+Discussion of the spectral filter design can be found in:
+JSH/TW Nodal DG Methods (DOI: 10.1007/978-0-387-72067-8), Section 5.3
```
Comment (MTCam) [inline mirgecom/filter.py:4]:
```suggestion
[Hesthaven_2009]_, Section. 5.3
```

**Comment by MTCam** [comment]
Thanks a whole bunch for breathing new life into this, btw!

**Review by inducer — APPROVED** [review]
Thanks! This looks great.

**Pull request merged**